src/controller.js

```diff
@@ -0,0 +1,105 @@
+(function exportController() {
+  class Controller {
+    constructor(ship) {
+      this.ship = ship;
+      this.initialiseSea();
+      document.querySelector('#sailbutton').addEventListener('click', () => this.setSail());
+    }
+
+    initialiseSea() {
+      const backgrounds = [
+        './images/water0.png',
+        './images/water1.png',
+      ];
+      let backgroundIndex = 0;
+      window.setInterval(() => {
+        document.querySelector('#viewport').style.backgroundImage = `url('${backgrounds[backgroundIndex % backgrounds.length]}')`;
+        backgroundIndex += 1;
+      }, 1000);
+    }
+
+    renderPorts(ports) {
+      const portsElement = document.querySelector('#ports');
+      portsElement.style.width = '0px';
+      ports.forEach((port, index) => {
+        const newPortElement = document.createElement('div');
+        newPortElement.className = 'port';
+        newPortElement.dataset.portName = port.portName;
+        newPortElement.dataset.portIndex = index;
+        portsElement.appendChild(newPortElement);
+        const portsElementWidth = parseInt(portsElement.style.width, 10);
```

Comment:
const and let should be declared at the start of the execution block {} (as above)